Objects/TI4_Helpers/TI4_ExploreHelper.ttslua

```diff
@@ -933,6 +1007,573 @@ function AttachLib._layoutDecals(decals)
+        local unitObject = getObjectFromGUID(unit.guid)
+        if unitObject then
+            graveyard.putObject(unitObject)
+            coroutine.yield(0)
+        end
+    end
+
+    -- Get owner tokens on planet, and send them to the graveyard
+    for _, ownerToken in ipairs(ownerTokensOnPlanet) do
+        local object = getObjectFromGUID(ownerToken)
+        if object then
+            graveyard.putObject(object)
+            coroutine.yield(0)
+        end
+    end
+
+    if purgeBag then
+        -- Get list of attachment tokens on planet, purge token and associated card
+        for _, attachTokenOnPlanet in ipairs(attachmentTokensOnPlanet) do
+            local attachTokenOnPlanetObject = getObjectFromGUID(attachTokenOnPlanet)
+            if attachTokenOnPlanetObject then
+                -- Does this token have an associated card?
+                local attachCardName = attachmentCardByToken[attachTokenOnPlanetObject.getName()]
+                if attachCardName then
+                    -- Did we find the card, and therefore have it's GUID?
+                    local attachCard = attachmentCardGuidByName[attachCardName]
+                    if attachCard then
+                        -- Is the card object still on the table?
+                        local attachCardObject = getObjectFromGUID(attachCard)
+                        if attachCardObject then
+                            purgeBag.putObject(attachCardObject)
+                            coroutine.yield(0)
+                        end
+                    end
+                end
+
+                attachTokenOnPlanetObject.setLock(false)
+                purgeBag.putObject(attachTokenOnPlanetObject)
+                coroutine.yield(0)
+            end
+        end
+
+        -- Get planet card, and purge it.
+        if planetCard then
+            local planetCardObject = getObjectFromGUID(planetCard)
+            if planetCardObject then
+                purgeBag.putObject(planetCardObject)
+                coroutine.yield(0)
+            end
+        end
+    end
+
+    -- Place token directly on planet, and lock it
+    attachTokenObject.setLock(false)
+    local planetPosition = systemObject.positionToWorld(planet.position)
+    local tokenPosition = { x = planetPosition.x, y = planetPosition.y + 0.5, z = planetPosition.z }
+    local collide = false
+    local fast = true
+    attachTokenObject.setPositionSmooth(tokenPosition, collide, fast)
+
+    local function condition()
+        if attachTokenObject.isSmoothMoving() then
+            return false
+        end
+        if not attachTokenObject.resting then
+            return false
+        end
+        return true
+    end
+    local function action()
+        attachTokenObject.setLock(true)
+    end
+    local function timeout() end
+    Wait.condition(action, condition, 3, timeout)
+
+    -- Clear context menu
+    attachTokenObject.clearContextMenu()
+
+    -- Inject planet with no stats
```

Comment:
Is this correct?  Shouldn't it remove the planet?

Reply:
I think that the number of planets can matter for visual reasons. It affects the number of zones in a system, at least. So this felt safer.